osu.Game/Screens/Play/SongProgressGraph.cs

```diff
@@ -6,43 +6,207 @@
 using System.Diagnostics;
 using osu.Game.Rulesets.Objects.Types;
 using osu.Game.Rulesets.Objects;
+using osu.Framework.Timing;
+using osu.Game.Beatmaps.Timing;
 
 namespace osu.Game.Screens.Play
 {
     public class SongProgressGraph : SquareGraph
     {
+        private List<double> strains = new List<double>();
+
+        public List<double> Strains
+        {
+            get { return strains; }
+            set
+            {
+                for(int x = 0; x < value.Count; x++)
+                {
+                    if (x == 0)
+                    {
+                        strains.Add(value[x]);
+                        strains.Add(value[x]);
+                        Values.Add(value[x]);
+                        Values.Add(value[x]);
+                    }
+                    else
+                    {
+                        strains.Add(value[x]);
+                        Values.Add(value[x]);
+                    }
```

Comment:
The `else` code block can be moved outside and only add the values again if `x == 0`.